demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,370 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    Field,
+    FieldSet,
+    FinalForm,
+    FinalFormInput,
+    FinalFormSaveSplitButton,
+    FinalFormSubmitEvent,
+    Loading,
+    MainContent,
+    TextField,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { ArrowLeft } from "@comet/admin-icons";
+import { EditPageLayout, queryUpdatedAt, resolveHasSaveConflict, useFormSaveConflict } from "@comet/cms-admin";
+import { IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createManufacturerMutation, manufacturerFormFragment, manufacturerQuery, updateManufacturerMutation } from "./ManufacturerForm.gql";
+import {
+    GQLCreateManufacturerMutation,
+    GQLCreateManufacturerMutationVariables,
+    GQLManufacturerFormDetailsFragment,
+    GQLManufacturerQuery,
+    GQLManufacturerQueryVariables,
+    GQLUpdateManufacturerMutation,
+    GQLUpdateManufacturerMutationVariables,
+} from "./ManufacturerForm.gql.generated";
+
+type FormValues = Omit<GQLManufacturerFormDetailsFragment, "address" | "addressAsEmbeddable"> & {
+    address: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
+    addressAsEmbeddable: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
+};
+
+interface FormProps {
+    id?: string;
+}
+
+export function ManufacturerForm({ id }: FormProps): React.ReactElement {
+    const stackApi = useStackApi();
+    const client = useApolloClient();
+    const mode = id ? "edit" : "add";
+    const formApiRef = useFormApiRef<FormValues>();
+    const stackSwitchApi = useStackSwitchApi();
+
+    const { data, error, loading, refetch } = useQuery<GQLManufacturerQuery, GQLManufacturerQueryVariables>(
+        manufacturerQuery,
+        id ? { variables: { id } } : { skip: true },
+    );
+
+    const initialValues = React.useMemo<Partial<FormValues>>(
+        () =>
+            data?.manufacturer
+                ? {
+                      ...filter<GQLManufacturerFormDetailsFragment>(manufacturerFormFragment, {
+                          ...data.manufacturer,
+                          address: data.manufacturer.address
```

Comment:
Is it a good idea to use `filter` on data that isn't GraphQL anymore?

Reply:
For example the fields inside `address` that are converted to numbers

Reply:
ah, I didn't see that he modifies the object before passing it into filter. No, that should be the other way around.

Reply:
alright. what's a better way to remove the __typename so that initialValues is compatible with graphql-input? maybe using spread-operator isn't the best solution here? as mentioned somewhere else regarding this change I don't know any js-daisy-chaining-trick removing only __typename...

Reply:
ah found it: https://github.com/vivid-planet/comet/pull/1751#discussion_r1503732940

Reply:
Doesn't filter remove the __typename? IMO it should, since it's not part of the fragment

Reply:
it does. but you said:

> Is it a good idea to use filter on data that isn't GraphQL anymore?

so use filter or something else?

Reply:
that's what I had before. the issue with this approach was with replacing values inside of objects without overwriting the other fields.

thinking about this I'm afraid I'll need a separate step to achieve this. because I need to object-merge any nested object with the filtered object which is currently directly spreaded (`...filter`).... I'll give it a try...

Reply:
@johnnyomair is changed.